Задача 3/Цк3-2/main.cpp

```diff
@@ -0,0 +1,50 @@
+// Task3-2.
+#include <iostream>
+#include <cmath>
+/**
+*\ brief  Рекурентное вычисление суммы первых n членов последовательности 
+*\ param  n - количество членов последовательности
+*\ return  сумма
+*/
+double sequenceRec(int n);
+/**
+*\ brief Рекурентное вычисление суммы первых членов последовательности не меньших заданного числа e
+*\ param e - заданное число
+*\ return сумма
+*/
+double sequenceRec2(double e);
+int main()
+{
+    setlocale(LC_ALL, "Russian");
+    int n;
+    double e;
+        std::cout << "Введите n " ; std::cin >> n;
+        std::cout << "Введите e "; std::cin >> e;
+        std::cout << "Сумма членов последовательности для n = " << n <<" "<< sequenceRec(n) << "\n";
+        std::cout << "Сумма членов последовательности для e = " << e << " " <<sequenceRec2(e) << "\n";
+    return 0;
+}
+double sequenceRec(int n)
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.